### PR TITLE
Document System.IO.StringReader

### PR DESCRIPTION
**Pull request by carlossanlop**
Code here: https://source.dot.net/#System.Runtime.Extensions/System/IO/StringReader.cs

### CODE REVIEW

File: xml/System.IO/StringReader.xml [hunk 1]
```diff
@@ -686,10 +687,19 @@ This implementation of `Close` calls the <xref:System.IO.StringReader.Dispose%2A
...
-        <remarks>To be added.</remarks>
+        <param name="buffer">When this method returns, contains the characters read from the current source. If the total number of characters read is zero, the span remains unmodified.</param>
+        <summary>Reads all the characters from the input string starting at the current position and advances the current position to the end of the input string.</summary>
+        <returns>The total number of characters read into the buffer.</returns>
+        <remarks>
+          <format type="text/markdown"><![CDATA[
+
+## Remarks
+
+<xref:M:System.IO.StringReader.ReadBlock(System.Span{System.Char})> internally calls <xref:M:System.IO.StringReader.Read(System.Span{System.Char})> directly.
```
Comment (gewarren) [inline xml/System.IO/StringReader.xml]:
```suggestion
<xref:System.IO.StringReader.ReadBlock(System.Span{System.Char})> internally calls <xref:System.IO.StringReader.Read(System.Span{System.Char})> directly.
```

File: xml/System.IO/StringReader.xml [hunk 2]
```diff
@@ -720,11 +730,20 @@ This implementation of `Close` calls the <xref:System.IO.StringReader.Dispose%2A
...
+        <param name="cancellationToken">The token to monitor for cancellation requests. The default value is <see cref="P:System.Threading.CancellationToken.None" />.</param>
+        <summary>Asynchronously reads all the characters from the input string starting at the current position and advances the current position to the end of the input string.</summary>
+        <returns>A task representing the asynchronous read operation. The value of the <paramref name="TResult" /> parameter contains the total number of characters read into the buffer.</returns>
+        <remarks>
+          <format type="text/markdown">
+  <![CDATA[
+
+## Remarks
+
+<xref:M:System.IO.StringReader.ReadBlockAsync(System.Memory{System.Char},System.Threading.CancellationToken)> calls <xref:M:System.IO.StringReader.ReadBlock(System.Span{System.Char})> asynchronously, which in turn calls <xref:M:System.IO.StringReader.Read(System.Span{System.Char})> directly.
```
Comment (gewarren) [inline xml/System.IO/StringReader.xml]:
```suggestion
<xref:M:System.IO.StringReader.ReadBlockAsync(System.Memory{System.Char},System.Threading.CancellationToken)> calls <xref:System.IO.StringReader.ReadBlock(System.Span{System.Char})> asynchronously, which in turn calls <xref:System.IO.StringReader.Read(System.Span{System.Char})> directly.
```

File: xml/System.IO/StringReader.xml (hunk 1 above)
Comment (carlossanlop) [reply xml/System.IO/StringReader.xml]: Nice catch, thanks.

**Comment by carlossanlop** [comment]
Thanks for your review, @gewarren. I applied all your suggestions. If all looks good, feel free to sign off whenever you get a chance.

**Pull request merged**